src/lib/logical_query_plan/abstract_lqp_node.hpp

```diff
@@ -144,6 +144,14 @@ class AbstractLQPNode : public std::enable_shared_from_this<AbstractLQPNode> {
   bool operator==(const AbstractLQPNode& rhs) const;
   bool operator!=(const AbstractLQPNode& rhs) const;
 
+  /**
+   *  Builds a hash code by hashing the node type combined with specific member variables of
+   *  derived classes (see _shallow_hash()) and the hash codes of non-empty input nodes recursively.
+   *  Node expressions are not taken into account since combining the hashes of the expressions
```

Comment:
But... node expressions are hashed!?

Reply:
Correct, I missed to update that comment.

Reply:
Correct, I missed to update this comment. Addressed in #1836